src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -2418,37 +2418,23 @@ internal bool IsConnectionAlive(bool throwOnException)
                 else
                 {
                     uint error;
+                    SniContext = SniContext.Snix_Connect;
 
-                    object readPacket = EmptyReadPacket;
-
-                    try
+                    error = CheckConnection();
+                    if ((error != TdsEnums.SNI_SUCCESS) && (error != TdsEnums.SNI_WAIT_TIMEOUT))
                     {
-                        SniContext = SniContext.Snix_Connect;
-
-                        error = CheckConnection();
-
-                        if ((error != TdsEnums.SNI_SUCCESS) && (error != TdsEnums.SNI_WAIT_TIMEOUT))
-                        {
-                            // Connection is dead
-                            isAlive = false;
-                            if (throwOnException)
-                            {
-                                // Get the error from SNI so that we can throw the correct exception
-                                AddError(_parser.ProcessSNIError(this));
-                                ThrowExceptionAndWarning();
-                            }
-                        }
-                        else
+                        // Connection is dead
+                        isAlive = false;
+                        if (throwOnException)
                         {
-                            _lastSuccessfulIOTimer._value = DateTime.UtcNow.Ticks;
+                            // Get the error from SNI so that we can throw the correct exception
+                            AddError(_parser.ProcessSNIError(this));
+                            ThrowExceptionAndWarning();
                         }
                     }
-                    finally
+                    else
                     {
-                        if (!IsPacketEmpty(readPacket))
-                        {
-                            ReleasePacket(readPacket);
-                        }
+                        _lastSuccessfulIOTimer._value = DateTime.UtcNow.Ticks;
```

Comment:
This one looks clearer in a side by side comparison, the diff makes it hard to understand. The readPacket variable is set to EmptyReadPacket never used and then IsPacketEmpty checked (always true) and ReleasedPacket'ed. I removed all those steps since the variable isn't used which means no allocation and thus no release. There is also no need for the exception handling because we don't have to guarentee the release anymore so the try is removed.

Reply:
I see. Thanks for the explanation.